assets/chat/js/messages.js

```diff
@@ -364,6 +370,56 @@ class ChatUserMessage extends ChatMessage {
     }
 }
 
+class ChatReplyMessage extends ChatUserMessage {
+    constructor(message, user, target, prevMessage, prevMessageId, messageId = null, timestamp = null) {
+        super(message, user, timestamp);
+        this.prevMessage = prevMessage;
+        this.prevMessageId = prevMessageId;
+        this.replytarget = target;
+        this.msgid = messageId ?? Date.now().toString() + Math.floor(Math.random() * 1000);
+        this.isown = false;
+        this.highlighted = false;
+        this.historical = false;
+        this.targetoutgoing = null;
+        this.tag = null;
+        this.slashme = false;
+        this.mentioned = [];
+    }
+    html(chat = null) {
+        const classes = [],
+            attr = {};
+        if (this.msgid) attr["data-msg-id"] = this.msgid;
+        if (this.user && this.user.username)
+            attr["data-username"] = this.user.username.toLowerCase();
+        if (this.mentioned && this.mentioned.length > 0)
+            attr["data-mentioned"] = this.mentioned.join(" ").toLowerCase();
+
+        if (this.isown) classes.push("msg-own");
+        if (this.slashme && !this.target && !this.targetoutgoing)
+            classes.push("msg-me");
+        if (this.historical) classes.push("msg-historical");
+        if (this.highlighted) classes.push("msg-highlight");
+        // Cuts the message limit into a quater
+        const prevPreview = this.prevMessage.length > 128 ? this.prevMessage.substring(0, 128) + "…" : this.prevMessage;
```

Comment:
can we use `white-space: nowrap; text-overflow: ellipsis` to make this responsive?